### PR TITLE
CP-2512 - Documentation for external facing courses support

### PR DESCRIPTION
**Pull request by kwstewar**
Only showing documentation for the one endpoint action we actually need to expose.

### CODE REVIEW

File: source/includes/_courses.md [hunk 1]
```diff
@@ -154,13 +154,11 @@ This endpoint retrieves all courses.
...
-filter[state] | `published` will display all courses that are in a published state <br> `unpublished` will display all courses that have not yet been published <br> `archived` will display all courses that have been archived
-page[number] | Courses are paginated, use this to choose which page you want
-page[size] | The default is `15` - How many courses are listed in each request/page
-tab | `published` will display all courses that are in a published state <br> `unpublished` will display all courses that have not yet been published <br> `archived` will display all courses that have been archived
+filter[partner_id] | Required. Scopes the courses returned to the partner specified.
+user_id | Specify a user id to get the courses that the user is enrolled in as a student. Defaults to the currently signed in user
+include_department_courses | If specified, courses in departments of `partner_id` that the user is enrolled in will also be returned
 sort | Orders the results based on the param. <br>Example: `name` - sort in alphabetical order
+page[number] | Courses are paginated, use this to choose which page you want
```
Comment (moomooskycow) [inline source/includes/_courses.md:161]: Oh snap does this mean we can paginate the CoursesPage now? And maybe try to refactor the learner_stats endpoint and its consumers to be paginated as well?

**Review by phrazzld — APPROVED** [review]

File: source/includes/_courses.md (hunk 1 above)
Comment (kwstewar) [reply source/includes/_courses.md:161]: Yep, I think the biggest bit of work will be updating our front-end table components to work with a paginated endpoint

Comment (moomooskycow) [reply source/includes/_courses.md:161]: Haven't looked too deep into it, but react-query has some pagination utilities built-in: https://react-query.tanstack.com/docs/guides/paginated-queries

**Comment by laila-vinson** [comment]
This looks good. Do you run a command to generate any of this, or is it all manual?

**Comment by kwstewar** [comment]
It's all manual

**Pull request merged**